p2p/host/relay/autorelay.go

```diff
@@ -90,105 +87,105 @@ func (h *AutoRelayHost) background(ctx context.Context) {
 
 	for {
 		wait := autonat.AutoNATRefreshInterval
-		switch h.autonat.Status() {
+		switch ar.autonat.Status() {
 		case autonat.NATStatusUnknown:
 			wait = autonat.AutoNATRetryInterval
 
 		case autonat.NATStatusPublic:
 			// invalidate addrs
-			h.mx.Lock()
-			if h.addrs != nil {
-				h.addrs = nil
+			ar.mx.Lock()
+			if ar.addrs != nil {
+				ar.addrs = nil
 				push = true
 			}
-			h.mx.Unlock()
+			ar.mx.Unlock()
 
 			// if we had previously announced relay addrs, push our public addrs
 			if push {
 				push = false
-				h.PushIdentify()
+				ar.host.PushIdentify()
 			}
 
 		case autonat.NATStatusPrivate:
 			push = false // clear, findRelays pushes as needed
-			h.findRelays(ctx)
+			ar.findRelays(ctx)
 		}
 
 		select {
-		case <-h.disconnect:
+		case <-ar.disconnect:
 			// invalidate addrs
-			h.mx.Lock()
-			if h.addrs != nil {
-				h.addrs = nil
+			ar.mx.Lock()
+			if ar.addrs != nil {
+				ar.addrs = nil
 				push = true
 			}
-			h.mx.Unlock()
+			ar.mx.Unlock()
 		case <-time.After(wait):
 		case <-ctx.Done():
 			return
 		}
 	}
 }
 
-func (h *AutoRelayHost) findRelays(ctx context.Context) {
-	h.mx.Lock()
-	if len(h.relays) >= DesiredRelays {
-		h.mx.Unlock()
+func (ar *AutoRelay) findRelays(ctx context.Context) {
+	ar.mx.Lock()
+	if len(ar.relays) >= DesiredRelays {
+		ar.mx.Unlock()
 		return
 	}
-	need := DesiredRelays - len(h.relays)
-	h.mx.Unlock()
+	need := DesiredRelays - len(ar.relays)
+	ar.mx.Unlock()
 
 	limit := 20
```

Comment:
can we bump this limit to 50? We have 25 relays deployed plus whatever user-run relays, so we should try to see them all.

Reply:
Pushed a commit that bumps that.